server/src/main/java/eu/fasten/server/FastenServer.java

```diff
@@ -219,77 +241,70 @@ public void run() {
-        }).collect(Collectors.toList());
+                    (localStorageDir != null) ? localStorageDir :
+                        "/mnt/fasten/local_storage/" + k.getClass().getSimpleName());
+            }).collect(Collectors.toList());
     }
 
     /**
      * Setup DB connection for DB plugins.
      *
      * @param dbPlugins list of DB plugins
      */
-    private void makeDBConnection(List<DBConnector> dbPlugins) {
-        dbPlugins.forEach((p) -> {
-            if (dbUrls != null) {
-                p.setDBConnection(dbUrls.entrySet().stream().map(e -> new AbstractMap.SimpleEntry<>(e.getKey(),
-                        e.getValue())).collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, e -> {
-                    try {
-                        logger.debug("Set {} DB connection successfully for plug-in {}",
-                                e.getKey(), p.getClass().getSimpleName());
-                        return getDSLContext(e.getValue());
-                    } catch (SQLException ex) {
-                        logger.error("Couldn't set {} DB connection for plug-in {}\n {} \n{}",
-                                e.getKey(), p.getClass().getSimpleName(), ex.getMessage(), ex.getStackTrace());
-                    }
-                    return null;
-                })));
+    private void registerDBConnections(List<DBConnector> dbPlugins) {
 
-            } else {
-                logger.error("Couldn't make a DB connection. Make sure that you have "
-                        + "provided a valid DB URL, username and password.");
+        if (dbUrls == null) {
+            logger.info("No DB connection info provided, skipping initialization.");
+            return;
+        }
+
+        Supplier<Map<String, DSLContext>> s = () -> {
+            var connectionPool = new HashMap<String, DSLContext>();
+            for (String configKey : dbUrls.keySet()) {
+                final var jdbcUrl = dbUrls.get(configKey);
+                final var connection = getDSLContext(jdbcUrl);
+                connectionPool.put(configKey, connection);
             }
-        });
+            return connectionPool;
+        };
+        dbPlugins.forEach(p -> p.setDBConnection(s.get()));
     }
 
     private void loadDependencyGraphResolvers(List<DependencyGraphUser> plugins) {
         plugins.forEach(p -> {
-            if (dbUrls != null && depGraphPath != null) {
-                DSLContext dbContext = null;
-                try {
-                    dbContext = getDSLContext(dbUrls.get("mvn"));
-                } catch (SQLException ex) {
-                    logger.error("Couldn't set {} DB connection for plug-in {}\n{}",
-                            dbUrls.get("mvn"), p.getClass().getSimpleName(), ex.getStackTrace());
-                }
-                p.loadGraphResolver(dbContext, depGraphPath);
-            } else {
+            if (dbUrls == null || depGraphPath == null) {
```

Comment:
Shouldn't it be necessary to end this if with a 'return' statement to prevent an NPE right after?

Reply:
As we discussed, I've added RuntimeException to prevent NPE